bundles/config/org.eclipse.smarthome.config.discovery/src/main/java/org/eclipse/smarthome/config/discovery/DiscoveryResult.java

```diff
@@ -80,12 +80,20 @@
      * discovered. Its actual value can be retrieved from the {@link DiscoveryResult#getProperties()} map. Such unique
      * identifiers are typically the <code>ipAddress</code>, the <code>macAddress</code> or the
      * <code>serialNumber</code> of the discovered thing.
-     * 
+     *
      * @return the representation property of this result object (could be null)
      */
     public String getRepresentationProperty();
 
     /**
+     * Returns the actual value of the representation property of the result object.
+     *
+     * @see #getRepresentationProperty()
+     * @return the value of the representation property of this result object (could be null)
+     */
+    public Object getRepresentationPropertyValue();
```

Comment:
I agree, there will be `discoveryResult.getProperties().get(discoveryResult.getRepresentationProperty()))` everywhere. The property name by itself is boring. Then let's rather change it and give it a crisp name. It will be API breaking, but I doubt anybody is using it 🤞